apps/03TicTacToe.js

```diff
@@ -23,22 +23,41 @@ function printBoard() {
 
 function horizontalWin() {
     // Your code here
+ return ( (board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn) ||
+          (board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn) ||
+          (board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn)  );
+          // console.log("horizontalWin");
 }
 
 function verticalWin() {
     // Your code here
+    return ( (board[0][0] === playerTurn && board[1][0] === playerTurn && board[2][0] === playerTurn) ||
+             (board[0][1] === playerTurn && board[1][1] === playerTurn && board[2][1] === playerTurn) ||
+             (board[0][2] === playerTurn && board[1][2] === playerTurn && board[2][2] === playerTurn)  );
 }
 
 function diagonalWin() {
     // Your code here
+    return ( (board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn) ||
+             (board[0][2] === playerTurn && board[1][1] === playerTurn && board[2][0] === playerTurn)  );
 }
 
 function checkForWin() {
     // Your code here
+    return (horizontalWin() || verticalWin() || diagonalWin());
 }
 
 function ticTacToe(row, column) {
-    // Your code here
+    // Places mark
+    board[row][column] = 'X';
```

Comment:
Your app works, but this line shouldn't be here. You shouldn't write 'X' into the row, column every turn.

This isn't a problem because it gets overwritten on line 58, but it is bad form. I'm giving you full credit, but please fix.